api/health/worker.go

```diff
@@ -204,10 +265,41 @@ func (w *worker) runCheck(ctx context.Context, wg *sync.WaitGroup, name string,
 		}
 
 		if prevResult.Error == nil {
-			w.metrics.failingChecks.Inc()
+			w.log.Warn("check started failing",
+				zap.String("namespace", w.namespace),
+				zap.String("name", name),
+				zap.Strings("tags", check.tags),
+				zap.Error(err),
+			)
+			if check.isApplicationCheck {
+				for tag := range w.tags {
+					w.metrics.failingChecks.WithLabelValues(tag).Inc()
+				}
+				w.numFailingApplicationChecks++
+			} else {
+				for _, tag := range check.tags {
+					w.metrics.failingChecks.WithLabelValues(tag).Inc()
+				}
+				w.metrics.failingChecks.WithLabelValues(AllTag).Inc()
```

Comment:
Why don't we do this in the `if` statement? Don't we want to increment the total number of failing checks ?

Reply:
Similar for decrementing this in the `else if prevResult.Error != nil` block below

Reply:
added comments explaining why we don't do it in the `if`